integration/spark/src/main/scala/org/apache/carbondata/indexserver/DistributedCountRDD.scala

```diff
@@ -96,20 +99,78 @@ class DistributedCountRDD(@transient ss: SparkSession, dataMapFormat: Distributa
     new DistributedPruneRDD(ss, dataMapFormat).partitions
   }
 
-  private def generateFuture(split: Seq[InputSplit])
+  private def generateFuture(split: Seq[InputSplit], globalQueue: SegmentProcessor)
     (implicit executionContext: ExecutionContext) = {
     Future {
-      val segments = split.map { inputSplit =>
+
+      var segmentsWorkStatus = split.map { inputSplit =>
         val distributable = inputSplit.asInstanceOf[DataMapDistributableWrapper]
         distributable.getDistributable.getSegment
           .setReadCommittedScope(dataMapFormat.getReadCommittedScope)
-        distributable.getDistributable.getSegment
+
+        val processedSegments = globalQueue.ifProcessSegment(distributable.getDistributable
+          .getSegment.getSegmentNo, dataMapFormat.getCarbonTable.getTableId)
+
+        val segmentWorkStatusList = new SegmentWorkStatus(distributable.getDistributable
+          .getSegment, !processedSegments)
+
+        // if ifprocesssegment = true, iswaiting = false
+        val processedSegmentsList = globalQueue.processSegment(segmentWorkStatusList,
+          dataMapFormat.getCarbonTable.getTableId)
+        segmentWorkStatusList
+      }
+
+      val queueSize = globalQueue.queueSize()
+      val getGlobalworkQueue = globalQueue.getGlobalWorkQueue
```

Comment:
Please remove all unused variables like this

Reply:
done